http-body/src/lib.rs

```diff
@@ -50,6 +50,26 @@ pub trait Body {
         cx: &mut Context<'_>,
     ) -> Poll<Option<Result<Frame<Self::Data>, Self::Error>>>;
 
+    /// Attempt to progress the body's state without pulling a new frame.
+    ///
+    /// `Body` consumers can use this method to allow the `Body` implementation to continue to
+    /// perform work even when the consumer is not yet ready to read the next frame. For example,
+    /// a `Body` implementation could maintain a timer counting down between `poll_frame` calls and
+    /// report an error from `poll_progress` when time expires.
+    ///
+    /// Consumers are *not* required to call this method. A `Body` implementation should not depend
```

Comment:
```suggestion
    /// Consumers are *not* required to call this method. A `Body` implementation must not depend
```

do you think it would be good to further strengthen this wording, replacing "_should_" with "_must_"?

Reply:
That might be a little strong - for example if you know your body is only going to be processed by hyper, and that hyper does use this API it could be reasonable to violate this recommendation. It's definitely true in the general case though.